src/main/java/com/azure/cosmos/examples/autoscalecontainercrud/async/AutoscaleContainerCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,210 @@
+                .contentResponseOnWriteEnabled(true)
+                .buildAsyncClient();
+
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+        updateContainerThroughput();
+
+        readContainerById();
+        readAllContainers();
+        // deleteAContainer() is called at shutdown()
+
+    }
+
+    // Database Create
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Create database " + databaseName + " if not exists...");
+
+        //  Create database if not exists
+        Mono<CosmosDatabaseResponse> databaseResponse = client.createDatabaseIfNotExists(databaseName);
+        database = client.getDatabase(databaseResponse.block().getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container create
+    private void createContainerIfNotExists() throws Exception {
+        logger.info("Create autoscale container " + containerName + " if not exists.");
+
+        // Container and autoscale throughput settings
+        CosmosContainerProperties autoscaleContainerProperties = new CosmosContainerProperties(containerName, "/lastName");
+        ThroughputProperties autoscaleThroughputProperties = ThroughputProperties.createAutoscaledThroughput(4000); //Set autoscale max RU/s
+
+        // Create the container with autoscale enabled
+        Mono<CosmosContainerResponse> databaseResponse = database.createContainer(autoscaleContainerProperties, autoscaleThroughputProperties,
+                new CosmosContainerRequestOptions());
+        container = database.getContainer(databaseResponse.block().getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Update container throughput
+    private void updateContainerThroughput() throws Exception {
+        logger.info("Update autoscale max throughput for container " + containerName + ".");
+
+        // Change the autoscale max throughput (RU/s)
+        container.replaceThroughput(ThroughputProperties.createAutoscaledThroughput(8000));
+
+        logger.info("Done.");
+    }
+
+    private void readContainerThroughput() throws Exception {
+        // Read the throughput on a resource
+        ThroughputProperties autoscaleContainerThroughput = container.readThroughput().block().getProperties();
+
+        // The autoscale max throughput (RU/s) of the resource
+        int autoscaleMaxThroughput = autoscaleContainerThroughput.getAutoscaleMaxThroughput();
+
+        // The throughput (RU/s) the resource is currently scaled to
+        //int currentThroughput = autoscaleContainerThroughput.Throughput;
+
+        //logger.info("Autoscale max throughput: {} current throughput: {}",autoscaleMaxThroughput,currentThroughput);
+    }
+
+    // Container read
+    private void readContainerById() throws Exception {
+        logger.info("Read container " + containerName + " by ID.");
+
+        //  Read container by ID
+        container = database.getContainer(containerName);
```

Comment:
database.getContainer(containerName) -> it is just a proxy, so there is no request being sent to server.

Reply:
was not 100% sure about this. I just added your description as a comment.